inference/core/roboflow_api.py

```diff
@@ -0,0 +1,198 @@
+from enum import Enum
+from typing import Any, Callable, Dict, Optional, Type, Union
+
+import requests
+from requests import Response
+
+from inference.core import logger
+from inference.core.entities.types import (
+    DatasetID,
+    ModelType,
+    TaskType,
+    VersionID,
+    WorkspaceID,
+)
+from inference.core.env import API_BASE_URL
+from inference.core.exceptions import (
+    MalformedRoboflowAPIResponseError,
+    MissingDefaultModelError,
+    RoboflowAPIConnectionError,
+    RoboflowAPINotAuthorisedError,
+    RoboflowAPINotNotFoundError,
+    RoboflowAPIUnsuccessfulRequestError,
+    WorkspaceLoadError,
+)
+from inference.core.utils.url_utils import wrap_url
+
+MODEL_TYPE_DEFAULTS = {
+    "object-detection": "yolov5v2s",
+    "instance-segmentation": "yolact",
+    "classification": "vit",
+}
+PROJECT_TASK_TYPE_KEY = "project_task_type"
+MODEL_TYPE_KEY = "model_type"
+
+
+def raise_from_lambda(
+    inner_error: Exception, exception_type: Type[Exception], message: str
+) -> None:
+    raise exception_type(message) from inner_error
+
+
+DEFAULT_ERROR_HANDLERS = {
+    401: lambda e: raise_from_lambda(
+        e,
+        RoboflowAPINotAuthorisedError,
```

Comment:
Another nit-picky comment: can we change Authorised to Authorized :-)

Reply:
👍